pkg/reconciler/mongodb/mongodbsource_test.go

```diff
@@ -74,179 +75,244 @@ func init() {
+						Database:   db,
+						Collection: coll,
+						Secret: corev1.LocalObjectReference{
+							Name: secretName,
+						},
+						SourceSpec: duckv1.SourceSpec{Sink: newSinkDestination()},
+					}),
+					WithMongoDbSourceUID(sourceUID),
+				),
+				newSink(),
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretName,
+						Namespace: testNS,
 					},
-					SourceSpec: duckv1.SourceSpec{Sink: newSinkDestination()},
-				}),
-				WithMongoDbSourceUID(sourceUID),
-				// Status Update:
-				WithInitMongoDbSourceConditions,
-				WithMongoDbSourceSink(sinkURI),
-				WithMongoDbSourceConnectionFailed("Unable to get MongoDb URI field"),
-			),
-		}},
-		WantEvents: []string{
-			Eventf(corev1.EventTypeWarning, "InternalError",
-				"Unable to get MongoDb URI field"),
-		},
-	}, {
-		Name:    "can't create mongodb client: does't start with mongodb://",
-		WantErr: true,
-		Objects: []runtime.Object{
-			NewMongoDbSource(sourceName, testNS,
-				WithMongoDbSourceSpec(sourcesv1alpha1.MongoDbSourceSpec{
-					Database:   db,
-					Collection: coll,
-					Secret: corev1.LocalObjectReference{
-						Name: secretName,
+					Data: map[string][]byte{
+						"URI": []byte("notValid"),
 					},
-					SourceSpec: duckv1.SourceSpec{Sink: newSinkDestination()},
-				}),
-				WithMongoDbSourceUID(sourceUID),
-			),
-			newSink(),
-			&corev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      secretName,
-					Namespace: testNS,
 				},
-				Data: map[string][]byte{
-					"URI": []byte("notValid"),
+			},
+			Key: testNS + "/" + sourceName,
+			OtherTestData: map[string]interface{}{
+				"mongo": mongotesting.TestClientData{
+					CreateClientErr: errors.New(`error parsing uri: scheme must be "mongodb" or "mongodb+srv"`),
 				},
 			},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: NewMongoDbSource(sourceName, testNS,
+					WithMongoDbSourceSpec(sourcesv1alpha1.MongoDbSourceSpec{
+						Database:   db,
+						Collection: coll,
+						Secret: corev1.LocalObjectReference{
+							Name: secretName,
+						},
+						SourceSpec: duckv1.SourceSpec{Sink: newSinkDestination()},
+					}),
+					WithMongoDbSourceUID(sourceUID),
+					// Status Update:
+					WithInitMongoDbSourceConditions,
+					WithMongoDbSourceSink(sinkURI),
+					WithMongoDbSourceConnectionFailed(`error parsing uri: scheme must be "mongodb" or "mongodb+srv"`),
+				),
+			}},
+			WantEvents: []string{
+				Eventf(corev1.EventTypeWarning, "InternalError",
+					`error parsing uri: scheme must be "mongodb" or "mongodb+srv"`),
+			},
 		},
-		Key: testNS + "/" + sourceName,
-		WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
-			Object: NewMongoDbSource(sourceName, testNS,
-				WithMongoDbSourceSpec(sourcesv1alpha1.MongoDbSourceSpec{
-					Database:   db,
-					Collection: coll,
-					Secret: corev1.LocalObjectReference{
-						Name: secretName,
+		{
+			Name:    "can't create mongodb client: does't start with mongodb://",
```

Comment:
nit: you can just call this UT something like error creating mongodb client... or something like that. It doesn't need to be related to doesn't start with mongodb://

you can do that in the follow up though